src/coredump/coredump.c

```diff
@@ -784,7 +784,6 @@ static int submit_coredump(
         if (journald_crash) {
                 /* We cannot log to the journal, so just print the message.
                  * The target was set previously to something safe. */
-                assert(startswith(core_message, "MESSAGE="));
```

Comment:
This is not useless. It checks that core_message does indeed start with the expected prefix. This is a check for a programming error.

Reply:
Given that we do `core_message = strjoina("MESSAGE=Process ", ...)` right before, I really don't see the point. I mean the string is built just before the assertion, it's not like it was built somewhere else and passed to this function.

Reply:
OK.